azure/pipelines/deploy/container-app.yml

```diff
@@ -0,0 +1,141 @@
+  displayName: 'Platina environment name.'
+- name: module
+  type: string
+  displayName: 'Platina module name.'
+- name: containerApp
+  type: string
+  displayName: 'Container application resource name.'
+- name: containerAppResourceGroup
+  type: string
+  displayName: 'Containe application resource group name.'
+- name: containerJsonPath
+  type: string
+  displayName: 'Path to the container.json file.'
+- name: label
+  type: string
+  displayName: 'Optional label to prefix steps in pipeline.'
+  default:
+- name: preDeploySteps
+  type: stepList
+  default: []
+  displayName: 'Optional steps to execute before deploying container app.'
+- name: postDeploySteps
+  type: stepList
+  default: []
+  displayName: 'Optional steps to execute after deploying container app.'
+
+steps:
+- task: PowerShell@2
+  displayName: '${{ coalesce(parameters.label, parameters.module) }} - Check preconditions'
+  inputs:
+    targetType: 'inline'
+    script: |
+      If (-Not (Test-Path '${{ parameters.containerJsonPath }}' -PathType Leaf)) {
+          Write-Host "'${{ parameters.containerJsonPath }}' file doesn't exist."
+          Write-Host "##vso[task.setVariable variable=skipContainerApp]true"
+      }
+      Else {
+          Write-Host "Preconditions have been met to continue container app deployment."
+          Write-Host "##vso[task.setVariable variable=skipContainerApp]false"
+      }
+
+- ${{ if ne(variables['skipContainerApp'], 'true') }}:
+  - ${{ each step in parameters.preDeploySteps }}:
+    - ${{ step }} 
+ 
+- task: AzureCLI@2
+  displayName: '${{ coalesce(parameters.label, parameters.module) }} - Pin image'
+  inputs:
+    azureSubscription: ${{ parameters.azureRegistrySubscription }} # Container registry is typically only available in production subscription.
+    scriptType: 'pscore'
+    scriptLocation: 'inlineScript'
+    inlineScript: |
+      $container = Get-Content "${{ parameters.containerJsonPath }}" | ConvertFrom-Json
+      $registry = $($container.registry)
+      $repository = $($container.repository)
+      $image = "$($repository):$($container.version)"
+      
+      # Allow implicit Azure CLI extension install (container app).
+      az config set extension.use_dynamic_install=yes_without_prompt
+      
+      # Tag image with a environment specific name so we keep track of the image that is deployed.
+      $pinnedImage = "$image-${{ parameters.environment }}"
+      Write-Host "##vso[task.setvariable variable=pinnedImage]$pinnedImage"
+     
+      Write-Host "Tagging $image as $pinnedImage"
+      az acr import --name $registry --source "$registry/$image" --image $pinnedImage --force
+
+      # Disallow deployed image modification/deletion so that it will not automatically cleaned up by purge tasks.
+      # In other words, lock the image.
+      Write-Host "Disallowing delete of $pinnedImage"
+      az acr repository update --name $registry --image $pinnedImage --write-enabled false --delete-enabled false
```

Comment:
I expected to have some commands in here to undo this action for images that are no longer relevant. Or is that done by simply removing the tag?

Reply:
Right, have added that before removing the tag